subtests/docker_cli/info/info.py

```diff
@@ -35,82 +33,32 @@ class info(subtest.Subtest):
-             Storage Driver: devicemapper
-              Pool Name: vg--docker-docker--pool
-              Pool Blocksize: 524.3 kB
-              ...
-             Logging Driver: journald
-
-        In this case 'Images', 'Server Version', and 'Logging Driver'
-        are simple tuples but 'Storage Driver' has both a value and
-        a set of further key/value pairs: Pool Name, Pool Blocksize, ...
-
-        We parse that and return a dict with the expected key/value
-        mapping *and* an extra: for all elements with subelements,
-        'element...' (element name plus three dots) is a dict containing
-        the subelements. E.g. x['Storage Driver...']['Pool Name'] = 'vg--etc'
-        """
-        out = {}
-        current_key = None
-        for line in cli_output.splitlines():
-            # Almost every line will be Foo: Bar, but 'Insecure Registries:'
-            # is followed by a simple list of IPv4 netmasks
-            if ': ' in line or line.endswith(':'):
-                (key, value) = [e.strip() for e in line.split(':', 1)]
-            else:
-                key = line.strip()
-                value = ''
-            if line.startswith(' '):
-                if not current_key:
-                    raise IndexError("Internal error: indented output line"
-                                     " '%s' from docker info with no previous"
-                                     " unindented lines.", line)
-                if current_key not in out:
-                    out[current_key] = {}
-                out[current_key][key] = value
-            else:
-                out[key] = value
-                current_key = key + '...'
-        return out
+        self.stuff['cmdresult'] = DockerInfo()
 
     def postprocess(self):
-        # Raise exception on Go Panic or usage help message
-        outputgood = OutputGood(self.stuff['cmdresult'])
-        info_map = self._build_table(outputgood.stdout_strip)
+        super(info, self).postprocess()
+        info_table = self.stuff['cmdresult']
 
         # We support multiple storage drivers. Each one has a different
         # set of key/value settings under 'info'; so each one has a
         # dedicated helper method for validating.
-        storage_driver = info_map['Storage Driver'].lower()
+        driver_name = info_table['Storage Driver']
+        if driver_name is None:
+            raise DockerTestFail("'docker info' did not return"
+                                 " a value for 'Storage Driver'")
+        self.loginfo("Storage Driver = %s", driver_name)
         try:
-            handler = '_postprocess_' + storage_driver
-            getattr(self, handler)(info_map['Storage Driver...'])
+            handler = '_postprocess_' + driver_name.lower()
+            getattr(self, handler)(info_table.sub_fields('Storage Driver'))
         except AttributeError:
-            raise DockerTestFail("Unknown storage driver: %s" % storage_driver)
-        except KeyError:
```

Comment:
I don't understand this. I removed the `KeyError` handler because I consider this possibility too unlikely, and not worth the clutter. If it ever happens IRL, the test will fail anyway. Does that address your concern?